redpanda/resources/serverlesscluster/resource_cluster.go

```diff
@@ -0,0 +1,251 @@
+
+// Package serverlesscluster contains the implementation of the ServerlessCluster resource
+// following the Terraform framework interfaces.
+package serverlesscluster
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	controlplanev1beta2 "buf.build/gen/go/redpandadata/cloud/protocolbuffers/go/redpanda/api/controlplane/v1beta2"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/cloud"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/config"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/models"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ resource.Resource                = &ServerlessCluster{}
+	_ resource.ResourceWithConfigure   = &ServerlessCluster{}
+	_ resource.ResourceWithImportState = &ServerlessCluster{}
+)
+
+// ServerlessCluster represents a cluster managed resource.
+type ServerlessCluster struct {
+	CpCl *cloud.ControlPlaneClientSet
+}
+
+// Metadata returns the full name of the ServerlessCluster resource.
+func (*ServerlessCluster) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "redpanda_serverless_cluster"
+}
+
+// Configure uses provider level data to configure ServerlessCluster's clients.
+func (c *ServerlessCluster) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	p, ok := req.ProviderData.(config.Resource)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *provider.Data, got: %T. Please report this issue to the provider developers.", req.ProviderData))
+		return
+	}
+
+	c.CpCl = cloud.NewControlPlaneClientSet(p.ControlPlaneConnection)
+}
+
+// Schema returns the schema for the ServerlessCluster resource.
+func (*ServerlessCluster) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resourceServerlessClusterSchema()
+}
+
+func resourceServerlessClusterSchema() schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required:      true,
+				Description:   "Name of the serverless cluster",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"serverless_region": schema.StringAttribute{
+				Optional:      true,
+				Description:   "Cloud provider specific region of the serverless cluster",
```

Comment:
It might be good to extend this with a doc link as I personally wasn't able to find one with a quick Google. Also, are the regions cloud provider specific (AWS/GCP) or Redpanda specific?